pkg/noun/nock.c

```diff
@@ -1117,16 +1119,23 @@ _n_bint(u3_noun* ops, u3_noun hif, u3_noun nef, c3_o los_o, c3_o tel_o)
         c3_w mem_w = 0;
         c3_y op_y;
 
-        // we just throw away the hint (why is this not a static hint?)
         tot_w += _n_comp(ops, hod, c3n, c3n);
         ++tot_w; _n_emit(ops, TOSS);
 
-        // memoizing code always loses TOS because SAVE needs [pro key]
         mem_w += _n_comp(&mem, nef, c3y, c3n);
         ++mem_w; _n_emit(&mem, SAVE);
 
         op_y   = (c3y == los_o) ? SLIB : SKIB; // overflows to SLIS / SKIS
-        ++tot_w; _n_emit(ops, u3nt(op_y, mem_w, u3k(nef)));
+        u3z_cid cid = u3z_memo_toss;
+        // skip over spot hints
+        // XX actually evaluate hod some day
+        while ( 11 == u3h(hod) && 1 == u3h(u3t(u3h(u3t(hod))))) {
```

Comment:
@frodwith if you take a look at the Nock produced from `!=  !:  ~>  %memo.[42]  42` you can see that `%spot` hints are wrapped and nested outside/within our `%memo` hint. The `while` loop here is meant to "ignore" those so we can parse the clue correctly.

Reply:
Without this logic, using the new `%memo` format (i.e., with a clue other than `[1 0]`) inside most vanes (i.e., the ones that use `!:` calls that wrap our `%memo` hints) doesn't work as intended. I'm open to a better algorithm here, but the code as written did solve the issue when I tested the new `%memo` hint in the same vanes I previously mentioned.

Reply:
It would be nice if you also skipped static hints in addition to dynamic hints with a [1 ...] clue when trying to resolve it statically. It would also be totally fine to ignore this hint when the clue is dynamic (modulo static hints). It's always fine to ignore (recursively) safe static hints. 

As an aside, it might be nice to make this a utility function. I haven't tried to compile this, but as a beginning:

```
/* extract a constant from a formula,
   ignoring safe/static hints,
   doing no computation */
u3_weak u3r_skip(u3_noun fol) {
  while ( 1 ) {
    if ( c3y == u3du(fol) ) {
      /* we -could- recurse on head and tail ... */
      return u3_none;
    }
    else switch ( u3h(fol) ) {
      default:
        /* we -could- try to constant fold... */
        return u3_none;
      case 1:
        return u3k(u3t(fol));
      case 11: {
        /* but we're only skipping hints here. */
        u3_noun arg = u3t(fol),
                hod = u3h(arg);
        
        if ( c3y == u3du(hod) ) {
          u3_weak dug = u3r_skip(u3t(hod));
          if ( u3_none == dug ) {
            /* only recursively safe dynamic ones. */
            return u3_none;
          }
          u3z(dug);
        }
        fol = u3t(arg);
        continue;
      }
    }
  }
}
```